### PR TITLE
feat(datastore): Add Datastore Admin API samples

### LINKED ISSUE #4120: Add Datastore Admin API samples
Add code samples for Datastore Admin API
[Generation configurations PR](https://github.com/googleapis/python-datastore/pull/27)

### PR DESCRIPTION
**Pull request by IlyaFaer**
Fixes #4120

### CODE REVIEW

**Comment by IlyaFaer** [comment]
@BenWhitehead, @crwilcox, I've used the code generated be Alex Vaskman (see the [generation configurations PR](https://github.com/googleapis/python-datastore/pull/27)), to run tests. They are okay:

Безымянный
We need an access for Datastore Admin and Storage simultaneously to write tests for entities export/import on Python and Go. I'll ask Deepika about this on our meeting today.

**Comment by BenWhitehead** [comment]
I think it's okay to not have tests for the export/import calls, since those are `LongRunningOperations` and depend on bucket access/config.

**Comment by tmatsuo** [comment]
@BenWhitehead 

I think we should add a test.

For bucket access and config, please tell us the requirements. We can set it up.
For long running operations, you can still test it. Wait for the result with a timeout that gives you around 90% success rate, then retry the test 2 or 3 times.

**Comment by BenWhitehead** [comment]
Hey @tmatsuo,

This is the quick list of what I believe would be needed to run import/export integration tests:
1. A storage bucket (ideally located in the same region is the datastore instance)
    * Some way of passing this bucket into the tests (env variable?)
2. Service Account used to run the tests will need
    * `Cloud Datastore Import Export Admin`, or `Cloud Datastore Owner` or `Owner` (least to most permissive)
    * `Storage Admin`, or `Owner` (least to most permissive)
3. The App Engine default service account of GCP Project used for integration tests needs to exist (it is the service account used by the backend jobs to do the import/export)
    * `Cloud Datastore Import Export Admin`, or `Cloud Datastore Owner` or `Owner` (least to most permissive)
    * `Storage Admin`, or `Owner` (least to most permissive) on the bucket from `1`


[Datastore Exporting and importing entities Permissions](https://cloud.google.com/datastore/docs/export-import-entities#permissions)

**Comment by tmatsuo** [comment]
@BenWhitehead 

> A storage bucket (ideally located in the same region is the datastore instance)

We're already using a storage bucket specified in `CLOUD_STORAGE_BUCKET` envvar. On Kokoro, this is set to `python-docs-samples-tests` bucket. I hope you can just use it.

Re: permission

I added those permissions to the service accounts.

I think `Storage Object Admin` is enough, instead of `Storage Admin`, so I only applied `Storage Object Admin` on the bucket to the app engine default service account. Let me know if we actually need `Storage Admin`.

**Comment by tmatsuo** [comment]
@IlyaFaer 

Can you also update the required env vars and permissions to run the tests in the README file? Actually you'll need to update README.rst.in and run the readmegen nox session to re-generate the README.rst file.

I think you can use `required_role` and `other_required_steps` field in that file ([Example usage](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/56a9d1ced54e907aabc19747cf749037f95d04f2/monitoring/api/v3/alerts-client/README.rst.in#L16)).

**Comment by IlyaFaer** [comment]
@BenWhitehead, @tmatsuo, looks like we were provided with all the necessary permissions in our project, thanks. I've run the tests locally - they're okay (I assume on kokoro they'll be failing until the next Datastore release):

Безымянный

I've unioned `import` and `export` tests as it's easier to run them sequentially and use the results of the export as an argument for import. Hope it's okay.

And, if I understood correctly, this `CLOUD_STORAGE_BUCKET` already exists in samples kokoro environment, so I don't need to create/delete it on every test run.

> Can you also update the required env vars and permissions to run the tests in the README file? Actually you'll need to update README.rst.in and run the readmegen nox session to re-generate the README.rst file

Taking a look

File: datastore/cloud-client/README.rst.in [hunk 1]
```diff
@@ -4,10 +4,16 @@ product:
...
   short_name: Cloud Datastore
   url: https://cloud.google.com/datastore/docs
-  description: > 
+  description: >
     `Google Cloud Datastore`_ is a NoSQL document database built for automatic
     scaling, high performance, and ease of application development.
 
+required_role: Datastore Import Export Admin
+other_required_steps: >
+    Set environment variables:
```
Comment (tmatsuo) [inline datastore/cloud-client/README.rst.in:13]: Don't you also need to add `Datastore Import Export Admin` and bucket permission to the App Engine default service account?

Comment (IlyaFaer) [reply datastore/cloud-client/README.rst.in:13]: I've added `Datastore Import Export Admin` into the `required_role` section. Pushing tip for `Storage Admin/Owner` roles.

Comment (tmatsuo) [reply datastore/cloud-client/README.rst.in:13]: There's no mention about App Engine default service account.

@BenWhitehead told me to add the permission to the App Engine default service account.

If it's required, I think we should mention in the README.

File: datastore/cloud-client/requirements-test.txt [hunk 2]
```diff
@@ -1,3 +1,4 @@
 backoff==1.10.0
 pytest==5.3.2
+retrying==1.3.3
```
Comment (tmatsuo) [inline datastore/cloud-client/requirements-test.txt:3]: retrying is an old library and not maintained. Can you use `backoff` instead?

File: datastore/cloud-client/README.rst.in (hunk 1 above)
Comment (BenWhitehead) [reply datastore/cloud-client/README.rst.in:13]: @IlyaFaer This is the outline of permissions we probably want to callout: https://github.com/GoogleCloudPlatform/python-docs-samples/pull/4121#issuecomment-647767343

File: datastore/cloud-client/requirements-test.txt (hunk 2 above)
Comment (IlyaFaer) [reply datastore/cloud-client/requirements-test.txt]: No problem, I've just taken it from another sample.

**Comment by leahecole** [comment]
just formally assigning this to @tmatsuo who has already been looking at it :)

File: datastore/cloud-client/admin.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    """
+    Exports a copy of all or a subset of entities from
+    Datastore to another storage system, such as Cloud Storage.
+    """
+    # project_id = "project-id"
+    # output_url_prefix = "gs://bucket-name"
+    client = datastore_admin_client.DatastoreAdminClient()
+
+    op = client.export_entities(project_id, output_url_prefix)
+    response = op.result()
```
Comment (tmatsuo) [inline datastore/cloud-client/admin.py:38]: `op.result()` can wait forever. Can we have a timeout?

File: datastore/cloud-client/admin.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+# [START datastore_admin_entities_import]
+def import_entities(project_id, input_url):
+    """Imports entities into Datastore."""
+    # project_id := "project-id"
+    # input_url := "gs://bucket-name/overall-export-metadata-file"
+    client = datastore_admin_client.DatastoreAdminClient()
+
+    op = client.import_entities(project_id, input_url)
+    response = op.result()
```
Comment (tmatsuo) [inline datastore/cloud-client/admin.py:53]: ditto

File: datastore/cloud-client/admin.py (hunk 3 above)
Comment (IlyaFaer) [reply datastore/cloud-client/admin.py]: No problem, pushed

**Comment by BenWhitehead** [comment]
[`google-cloud-datastore v1.13.0`](https://github.com/googleapis/python-datastore/releases/tag/v1.13.0) which includes the admin client has been released. We should be able to bump the `requirements.txt` and get things fixed up to merge.

**Comment by IlyaFaer** [comment]
@BenWhitehead, I don't see `gapic` directory in the package. There should be several files, e.g. `datastore_admin_v1/gapic/datastore_admin_client.py`, they are listed in ["add datastore admin client" PR](https://github.com/googleapis/python-datastore/pull/39/files) (this PR is mentioned in the release notes), however, it looks like they are missing in the Python package (so we're still getting the same error in kokoro checks)

The interesting thing is that I've downloaded the zip archive of the release, and `gapic` is there.

**Comment by crwilcox** [comment]
Shipped 1.13.1 with init.py and types.py that were missing. @IlyaFaer is this looking better now?

**Comment by IlyaFaer** [comment]
Thanks, @crwilcox! All the files are there now. I suppose only [renovate PR](https://github.com/GoogleCloudPlatform/python-docs-samples/pull/4295) is left, and kokoro checks will become green.

**Comment by tmatsuo** [comment]
@IlyaFaer FYI, I think the test will automatically executed if you join googlecloudplatform org.

**Comment by BenWhitehead** [comment]
@IlyaFaer looks like there are some test bootstrapping issues

```
==================================== ERRORS ====================================
____________ ERROR collecting datastore/cloud-client/admin_test.py _____________
ImportError while importing test module '/workspace/datastore/cloud-client/admin_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
admin_test.py:19: in <module>
    import admin
admin.py:15: in <module>
    from google.cloud.datastore_admin_v1.gapic import datastore_admin_client
E   ModuleNotFoundError: No module named 'google.cloud.datastore_admin_v1.gapic'
----- generated xml file: /workspace/datastore/cloud-client/sponge_log.xml -----
=========================== short test summary info ============================
ERROR admin_test.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.51s ===============================
```

**Comment by IlyaFaer** [comment]
@BenWhitehead, @crwilcox, I've updated the branch with renovate PR changes. Now another error appeared:
```python
____________ ERROR collecting datastore/cloud-client/admin_test.py _____________
Traceback (most recent call last):
  File "/workspace/datastore/cloud-client/admin_test.py", line 19, in <module>
    import admin
  File "/workspace/datastore/cloud-client/admin.py", line 15, in <module>
    from google.cloud.datastore_admin_v1.gapic import datastore_admin_client
  File "/workspace/datastore/cloud-client/.nox/py-3-6/lib/python3.6/site-packages/google/cloud/datastore_admin_v1/__init__.py", line 23, in <module>
    from google.cloud.datastore_admin_v1.gapic import datastore_admin_client
  File "/workspace/datastore/cloud-client/.nox/py-3-6/lib/python3.6/site-packages/google/cloud/datastore_admin_v1/gapic/datastore_admin_client.py", line 48, in <module>
    "google-cloud-datastore-admin",
  File "/workspace/datastore/cloud-client/.nox/py-3-6/lib/python3.6/site-packages/pkg_resources/__init__.py", line 481, in get_distribution
    dist = get_provider(dist)
  File "/workspace/datastore/cloud-client/.nox/py-3-6/lib/python3.6/site-packages/pkg_resources/__init__.py", line 357, in get_provider
    return working_set.find(moduleOrReq) or require(str(moduleOrReq))[0]
  File "/workspace/datastore/cloud-client/.nox/py-3-6/lib/python3.6/site-packages/pkg_resources/__init__.py", line 900, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/workspace/datastore/cloud-client/.nox/py-3-6/lib/python3.6/site-packages/pkg_resources/__init__.py", line 786, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'google-cloud-datastore-admin' distribution was not found and is required by the application
```
I suspect the problem caused with the generated lines:
https://github.com/googleapis/python-datastore/blob/592c7aa636990533a9c02f73a413de44b71dcc98/google/cloud/datastore_admin_v1/gapic/datastore_admin_client.py#L47-L49

There are no such a Python package `google-cloud-datastore-admin`.

**Comment by IlyaFaer** [comment]
@tmatsuo, @crwilcox, @BenWhitehead, all the checks are green :tada:
Thanks for the re-generation and dependencies

**Pull request merged**